### ISSUE TITLE
False positive "Code after RETURN statement cannot be executed"

### ISSUE THREAD

**Issue by bancer** [cakephp/cakephp-codesniffer#213]
```
	protected function _setProperty(?int $param) {
		if (!empty($param)) {
			$this->propery = $param;
		}
	}

	public function testMethod($param) {
		switch ($param) {
			case 'a':
				return 1;
			default:
				return 0;
		}
	}
```
this code produces invalid warning:
```
 285 | WARNING | Code after RETURN statement cannot be executed
```

It seems that `?int` in `_setProperty` method is the cause of the warning in `testMethod`.

PHP 7.1.18
CakePHP 2.10.10
PHP Code Sniffer 1.5.6
CakePHP Code Sniffer 1.0.4

**Comment by dereuromark** [comment]
Sure this is not coming from the phpcs sniffer package (not our sniffer), as such you should open a ticket there then.

**Comment by bancer** [comment]
https://github.com/squizlabs/PHP_CodeSniffer/issues/1995

According to this issue they do not support that old version of code sniffer anymore :(.

**Comment by markstory** [comment]
@bancer Given Greg's response I'm not sure there is much we can do either. I personally have no interest/time in upgrading the 2.x code sniffer rules to work with new phpcs.

**Comment by bancer** [comment]
Maybe @nojimage will be willing to take a look? He recently fixed two similar issues.

**Comment by dereuromark** [comment]
Is there sth to be done here? People should be using 3.x of code sniffer by now.

**Comment by markstory** [comment]
Its been more than a year of this issue being open. I'm going to close it. If folks feel really strongly they can comment here and provide a working patch.
